sympy/polys/tests/test_heuristicgcd.py

```diff
@@ -139,3 +141,12 @@ def test_issue_10996():
 
     assert H == 12*x**3*y**4 - 3*x*y**6 + 12*y**2*z
     assert H*cff == f and H*cfg == g
+
+
+def test_issue_25793():
+    R, x = ring("x", ZZ)
+    f = x - 4851  # failure starts for values more than 4850
+    g = f*(2*x + 1)
+    H, cff, cfg = heugcd(f, g)
+    assert H == f
```

Comment:
This test already passes on master because it is using the sparse implementation which was already fixed in gh-18566.

Reply:
Thanks -- updated and tested that failure occurs, now.